src/EnergyPlus/ScheduleManager.cc

```diff
@@ -690,28 +690,34 @@ namespace ScheduleManager {
 				ErrorsFound = true;
 			}
 
-			ProcessIntervalFields( Alphas( { 4, _ } ), Numbers, NumFields, NumNumbers, MinuteValue, SetMinuteValue, ErrorsFound, Alphas( 1 ), CurrentModuleObject, (Alphas( 3 ) == "YES") );
 			// Depending on value of "Interpolate" field, the value for each time step in each hour gets processed:
-			if ( Alphas( 3 ) != "NO" && Alphas( 3 ) != "YES" ) {
+			if ( SameString( Alphas( 3 ), "NO" ) ) {
+				DaySchedule( Count ).IntervalInterpolated = ScheduleInterpolation::No;
+			} else if ( SameString( Alphas( 3 ), "AVERAGE" ) ) {
+				DaySchedule( Count ).IntervalInterpolated = ScheduleInterpolation::Average;
+			} else if ( SameString( Alphas( 3 ), "LINEAR" ) ) {
+				DaySchedule( Count ).IntervalInterpolated = ScheduleInterpolation::Linear;
+			} else {
+				DaySchedule( Count ).IntervalInterpolated = ScheduleInterpolation::No;
```

Comment:
Is there a reason to assign here if you are aborting anyway?

Reply:
Not really, just a programming habit to always set the variable when having a "case" structure.